### DISCUSSION TITLE
How to decode two formats into one enum

### DISCUSSION THREAD

**Discussion by dolfs** [SwiftyLab/MetaCodable#115] [General]
I have a REST response that comes in one of two format, wanting me to have an enum like this:
```
public enum Items<T: Codable>: Codable {
    case list([T])
    case paginated(PaginationInfo, [T])
}
```
The corresponding JSON for the first case corresponds to a [[String: T]] (and I have been able to create a HelperCode to transform that to [T]. The second case looks like a dictionary where one of the keys (different names for different T.type) has to above list value, and there may be several other keys that I would want to decode into the PaginationInfo (it has some optional values).
The problem I am having is that currently, when using @Codable on the enum, it wants to look for key names "list" and "paginated." How do you use MetaCodable to do this as described?
An alternative, if the above is not possible, is to use:
```
structure Items {
    let paginationInfo: PaginationInfo?
    let items: [T]
}
```
but, due to the formats returned, all item lists will have a different key (in other words looking for "items") will not work. I know I can also not use MetaCodable and write a completely custom init(from:).

**Comment by soumyamahunt** [comment]
> The problem I am having is that currently, when using @Codable on the enum, it wants to look for key names "list" and "paginated." How do you use MetaCodable to do this as described?

The default behaviour for MetaCodable with enums is to look for identifier distinguishing the case. You can opt out of any identifiers by using [UnTagged](https://swiftpackageindex.com/swiftylab/metacodable/main/documentation/metacodable/untagged()) macro

> An alternative, if the above is not possible, is to use:
> 
> ```
> 
> structure Items {
> 
>     let paginationInfo: PaginationInfo?
> 
>     let items: [T]
> 
> }
> 
> ```
> 
> but, due to the formats returned, all item lists will have a different key (in other words looking for "items") will not work.

You can opt out of keyed decoding of some fields for structs as well by passing no arguments to [CodedAt](https://swiftpackageindex.com/swiftylab/metacodable/main/documentation/metacodable/codedat(_:)) macro for that field. [Here](https://swiftpackageindex.com/swiftylab/metacodable/main/tutorials/metacodable/essential#Share-common-Codable-fields) is a sample example.
